lib/environments/meteorEnvironment.js

```diff
@@ -276,24 +276,16 @@ function extractExportsFromMeteorPackage(
     }
 
     buildIsopack.declaredExports.forEach((declaredExport: Object) => {
-      if (declaredExport.testOnly) {
-        return;
+      if (!declaredExport.testOnly) {
+        declaredExports.add(declaredExport.name);
       }
-      if (declaredExports.indexOf(declaredExport.name) !== -1) {
-        return;
-      }
-
-      // This declaredExport is not a "testOnly" export and is not already
-      // amongst those we've gathered, so we want to push it into our
-      // collection.
-      declaredExports.push(declaredExport.name);
     });
 
     // TODO: If the "resources" section of the buildIsopack specifies a
     // mainModule, we need to attempt to scan it to find exports.
   });
 
-  return declaredExports;
+  return Array.from(declaredExports);
```

Comment:
In trying to use Sets more myself, I ran into the problem that Flow has a bug causing it to errantly complain about a type mismatch with the normal [...declaredExports] syntax. I'm guessing you ran into it too. I've found that an explicit cast gets around the problem, i.e. `[...(declaredExports: any)]`. A true fix is in the works in Flow.